docs/alfred.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import os
+from collections import namedtuple
+import asyncio
+from tqdm.asyncio import tqdm
+import typing as t
+from langchain_openai.chat_models import ChatOpenAI
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain.prompts import ChatPromptTemplate
+
+File = namedtuple("File", "name content")
+
+
+def get_files(path: str, ext: str) -> list:
+    return [os.path.join(path, f) for f in os.listdir(path) if f.endswith(ext)]
+
+
+def load_docs(path: str) -> t.List[File]:
+    files = [*get_files(path, ".md")]
+    docs = []
+    for file in files:
+        with open(file, "r") as f:
+            docs.append(File(file, f.read()))
+    return docs
+
+
+async def fix_doc_with_llm(doc: File, llm: BaseChatModel) -> File:
+    prompt = """\
+fix the following grammar and spelling mistakes in the following text. 
+Please keep the markdown format intact when reformating it. 
+Do not make any change to the parts of text that are for formating or additional metadata for the core text in markdown.
+The target audience for this is developers so keep the tone serious and to the point without any marketing terms. 
+The output text should me in .md format. 
+
+text: {text}
+"""
```

Comment:
added a script that helps review docs too

Reply:
I've noticed that gpt4 actually helps a lot in keeping the tone professional and consistent.

Reply:
but I don't think the prompt is proper - can be improved